server/server/managers.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+import json
+import uuid
+from typing import Dict, List
+
+from fastapi import WebSocket
+
+import server.user as user
+
+
+class DbManager:
+    """Manages the Database operations"""
+
+    def __init__(self, user_db_file: str, rooms_db_file: str):
+        self.user_db_file = user_db_file
+        self.rooms_db_file = rooms_db_file
+        u = open(user_db_file)
+        r = open(rooms_db_file)
+        self.users = json.loads(u.read())
+        self.rooms = json.loads(r.read())
+        u.close()
+        r.close()
+        print("entering")
+
+    def get_user(self, user_id: str = "") -> Dict:
+        """Fetches the user data from Database"""
+        if user_id:
+            return self.users.get(user_id, None)
+        else:
+            return self.users
+
+    def get_user_rooms(self, user_id: str) -> List:
+        """Fetches the room data for a user from Database"""
+        selected_rooms = []
+        for i in self.rooms:
+            if user_id in i:
+                selected_rooms.append(self.rooms[i])
+        return selected_rooms
+
+    def create_room(self, sender_id: str, reciever_id: str) -> str:
+        """Creates a new room(only if the persons don't already have one)"""
+        room_id = sender_id + reciever_id
+        if ((sender_id + reciever_id) in self.rooms) or (
+            (reciever_id + sender_id) in self.rooms
+        ):
+            return None
+        self.rooms[room_id] = {
+            "room_id": room_id,
+            "users": [sender_id, reciever_id],
+            "messages": [],
+        }
+        return room_id
```

Comment:
save room to json when ever it is updated, dont wait for server to close as that may not save it in case of crashes

Reply:
okay, i will write a flush function